modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/Ref.kt

```diff
@@ -110,6 +110,13 @@ interface Ref<F, A> {
       unsafe(a, MD)
     }
 
+    /**
+     * Creates an asynchronous, concurrent mutable reference initialized using the supplied function.
+     */
+    fun <F, A> of(MD: MonadDefer<F>, f: () -> A): Kind<F, Ref<F, A>> = MD.delay {
```

Comment:
Why should also add this factory in MonadDefer to create new refs without the argument. That way in fx we can just create refs from the this scope

Reply:
Also for all others that are constructed similarly

Reply:
We need to do this for MD/Async & Concurrent for cancelation.